File: binance4j-connectors/src/test/java/com/binance4j/connectors/RestConnectorsTest.java
package com.binance4j.connectors;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;

import org.junit.jupiter.api.Test;

import com.binance4j.core.client.RestClient;
import com.binance4j.core.exception.ApiException;
import com.binance4j.core.test.CustomTest;

public class RestConnectorsTest extends CustomTest {
	Connectors connectors;

	public RestConnectorsTest() {
		connectors = new Connectors(key, secret);
	}

	@Test
	void testGetBLVTClient() throws ApiException {
		assertNotNull(connectors.rest().blvt());
	}

	@Test
	void testGetC2CClient() {
		assertNotNull(connectors.rest().c2c());
	}

	@Test
	void testGetFiatClient() {
		assertNotNull(connectors.rest().fiat());
	}

	@Test
	void testGetLoanClient() {
		assertNotNull(connectors.rest().loan());
	}

	@Test
	void testGetMarginClient() {
		assertNotNull(connectors.rest().margin());
	}

	@Test
	void testGetMarketClient() {
		assertNotNull(connectors.rest().market());
	}

	@Test
	void testGetMiningClient() {
		assertNotNull(connectors.rest().mining());
	}

	@Test
	void testGetNFTClient() {
		assertNotNull(connectors.rest().nft());
	}

	@Test
	void testGetPayClient() {
		assertNotNull(connectors.rest().pay());
	}

	@Test
	void testGetPortfolioMarginClient() {
		assertNotNull(connectors.rest().margin());
	}

	@Test
	void testGetRebateClient() {
		assertNotNull(connectors.rest().rebate());
	}

	@Test
	void testGetSavingsClient() {
		assertNotNull(connectors.rest().savings());
	}

	@Test
	void testGetSpotClient() {
		assertNotNull(connectors.rest().spot());
	}

	@Test
	void testGetStakingClient() {
		assertNotNull(connectors.rest().staking());
	}

	@Test
	void testGetVisionSpotClient() {
		assertNotNull(connectors.rest().visionSpot());
	}

	@Test
	void testGetWalletClient() {
		assertNotNull(connectors.rest().wallet());
	}

	@Test
	void testUpdateKeys() {
		assertEquals(connectors.rest().getKey(), connectors.rest().blvt().getKey());
		assertEquals(connectors.rest().getSecret(), connectors.rest().blvt().getSecret());
		connectors.rest().updateKeys("foo", "bar");
		assertEquals(connectors.rest().getKey(), connectors.rest().blvt().getKey());
		assertEquals(connectors.rest().getSecret(), connectors.rest().blvt().getSecret());
	}

	@Override
	protected RestClient<?> getClient() {
		// TODO Auto-generated method stub
		return null;
	}
}

File: connectors/src/test/kotlin/com/binance4j/connectors/RestConnectorsTest.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.connectors

import com.binance4j.core.test.CustomTest
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class RestConnectorsTest : CustomTest() {
	@Test
	fun testGetBLVTClient() {
		Assertions.assertNotNull(Connectors.rest(key, secret).blvt())
	}
	
	@Test
	fun testGetC2CClient() {
		Assertions.assertNotNull(Connectors.rest().c2c())
	}
	
	@Test
	fun testGetFiatClient() {
		Assertions.assertNotNull(Connectors.rest().fiat())
	}
	
	@Test
	fun testGetLoanClient() {
		Assertions.assertNotNull(Connectors.rest().loan())
	}
	
	@Test
	fun testGetMarginClient() {
		Assertions.assertNotNull(Connectors.rest().margin())
	}
	
	@Test
	fun testGetMarketClient() {
		Assertions.assertNotNull(Connectors.rest().market())
	}
	
	@Test
	fun testGetMiningClient() {
		Assertions.assertNotNull(Connectors.rest().mining())
	}
	
	@Test
	fun testGetNFTClient() {
		Assertions.assertNotNull(Connectors.rest().nft())
	}
	
	@Test
	fun testGetPayClient() {
		Assertions.assertNotNull(Connectors.rest().pay())
	}
	
	@Test
	fun testGetPortfolioMarginClient() {
		Assertions.assertNotNull(Connectors.rest().margin())
	}
	
	@Test
	fun testGetRebateClient() {
		Assertions.assertNotNull(Connectors.rest().rebate())
	}
	
	@Test
	fun testGetSavingsClient() {
		Assertions.assertNotNull(Connectors.rest().savings())
	}
	
	@Test
	fun testGetSpotClient() {
		Assertions.assertNotNull(Connectors.rest().spot())
	}
	
	@Test
	fun testGetStakingClient() {
		Assertions.assertNotNull(Connectors.rest().staking())
	}
	
	@Test
	fun testGetVisionSpotClient() {
		Assertions.assertNotNull(Connectors.rest().visionSpot())
	}
	
	@Test
	fun testGetWalletClient() {
		Assertions.assertNotNull(Connectors.rest().wallet())
	}
	
	@Test
	fun testUpdateKeys() {
		assertEquals(Connectors.rest().key, Connectors.rest().c2c().key)
		assertEquals(Connectors.rest().secret, Connectors.rest().blvt().secret)
		Connectors.rest().updateKeys("foo", "bar")
		assertEquals(Connectors.rest().key, Connectors.rest().convert().key)
		assertEquals(Connectors.rest().secret, Connectors.rest().blvt().secret)
	}
}